syncd/FlexCounter.cpp

```diff
@@ -995,10 +1080,41 @@ class AttrContext : public CounterContext<AttrType>
                 values.emplace_back(meta->attridname, sai_serialize_attr_value(*meta, attrs[i]));
             }
             countersTable.set(sai_serialize_object_id(vid), values, "");
+            if (switchType == "voq")
+            {
+               // Chassis based system
+               appDbCountersTable->set(sai_serialize_object_id(vid), values, "");
+            }
         }
     }
 };
 
+void FlexCounter::getCfgSwitchType(
+         _In_ std::string &switch_type)
+{
+   try
+   {
+      if (!m_cfgDeviceMetaDataTable->hget("localhost", "switch_type", switch_type))
```

Comment:
is this connecting for localhost database to get this ? this seems a hacky way to do that, also all variables are hard coded, those should be defined somewhere with comments

Reply:
Pretty much every access to database is hardcoded in FlexCounter.cpp. We probably should change all of them in a separate pull request to not mix things up.